### PR TITLE
Harpy Species

### PR DESCRIPTION
**Pull request by VMSolidus**
## About the PR
This PR adds a new species to the game, that of the Harpy. Harpies are an avian humanoid species, that features colorful feathered wings+tail, and can mimick nearly any sound. 

Harpies have the following mechanical traits:
(+) They have slashing based melee, in the form of their talons
(+/=) They can imitate a huge variety of noises, both organic and inorganic in nature. This includes anything from a PDA ringtone, to a pew pew laser, or even sounds that normally come from other species. 
(+/=) They can natively play midis, effectively imitating entire songs. 
(=) They are slightly smaller than a normal human
(-) They take %15 more airloss damage
(-) They take %15 more brute damage

## Why / Balance
Harpies are made by special request to produce content uniquely intended for DeltaV. Although I don't remember who exactly, but I was told that "Monster People" races was generally wanted by the playerbase. Thus this entire PR has been a collaborative project entirely by individuals from the DeltaV discord. I promise of course that I fully intend to do Kitsune next, and I may even have an actual idea on how their fox transforming mechanic might be done with minimal new code. 

## Technical details

A single new YAML component has been added to the game, called type: HarpySinger. It can be added to any entity that a player might feasibly control, and allows it to use the same function PAI has for playing midis. Singer requires additional components in the form of Instrument(to declare what "Instrument" the entity is intended to use), and UserInterface to produce the necessary UI button. 

## Media
birdgoespew

## Credits
@jj4321 - Coding assistance with the Singer component
@debugok - Coding assistance with the Singer component
@leonardo_dabepis - Spritework. All character sprites 
@justanorang - Spritework, doing the UI component for singing
@DEATHB4DEFEAT - Graciously allowing me to use harpy.png from Simplestation, you're awesome!

**Changelog**
:cl: VMSolidus
- add: Harpies have been added to the game as a new playable species!

### CODE REVIEW

File: Resources/Locale/en-US/deltav/store/uplink-catalog.ftl.txt
Comment (FluffiestFloof) [inline Resources/Locale/en-US/deltav/store/uplink-catalog.ftl.txt:1]: It probably should be a  `.ftl.txt` but a `.ftl`

**Comment by evilexecutive** [comment]
I'm undrafting this so I can let the Linters do their thing and validate my work.

**Comment by DEATHB4DEFEAT** [comment]
> I'm undrafting this so I can let the Linters do their thing and validate my work.

You can run most of the tests locally using these commands:
```cs
dotnet restore
dotnet build --configuration DebugOpt --no-restore /p:WarningsAsErrors=nullable /m
dotnet run --project Content.YAMLLinter/Content.YAMLLinter.csproj --no-build -- NUnit.ConsoleOut=0
dotnet test --no-build --configuration DebugOpt Content.Tests/Content.Tests.csproj -- NUnit.ConsoleOut=0
dotnet test --no-build --configuration DebugOpt Content.IntegrationTests/Content.IntegrationTests.csproj -- NUnit.ConsoleOut=0 NUnit.MapWarningTo=Failed
```

**Comment by evilexecutive** [comment]
Oh I appreciate it Death! Can you mark this back as a draft then? I'll run them locally

**Comment by evilexecutive** [comment]
ALRIGHT LET'S GO

**Comment by evilexecutive** [comment]
Alright great, all bugs with THIS pr are fixed, harpies are now ready to go!

**Comment by evilexecutive** [comment]
#2 
New Custom Species added to Rebase. That makes 3 now, with Felinids, Vulpakin, and Harpies. I'll start work in the future for Kitsune, but I don't intend to have Kitsune ready for Rebase Beta.

**Comment by Colin-Tel** [comment]
To me, it looks good as far as the files are concerned. I'll open it up in-game and take a look at it.

**Comment by Colin-Tel** [comment]
image
get a load of this guy

**Comment by Colin-Tel** [comment]
image
Thank God that's outta the way 😏

**Review by Colin-Tel — APPROVED** [review]
I wanna pat the harpy on the head.

**Comment by Carolyn3114** [comment]
> I wanna pat the harpy on the head.

which head?

**Comment by VMSolidus** [comment]
It's like returning to where I was created.
https://github.com/Simple-Station/Einstein-Engines/pull/2571

**Pull request merged**